plumbing/transport/file/client.go

```diff
@@ -96,7 +97,23 @@ func (r *runner) Command(cmd string, ep *transport.Endpoint, auth transport.Auth
 		}
 	}
 
-	return &command{cmd: execabs.Command(cmd, ep.Path)}, nil
+	return &command{cmd: execabs.Command(cmd, adjustPathForWindows(ep.Path))}, nil
+}
+
+func isDriveLetter(c byte) bool {
+	return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z')
+}
+
+// On Windows, the path that results from a file: URL has a leading slash. This
+// has to be removed if there's a drive letter
+func adjustPathForWindows(p string) string {
+	if runtime.GOOS != "windows" {
+		return p
+	}
+	if len(p) >= 3 && p[0] == '/' && isDriveLetter(p[1]) && p[2] == ':' {
+		return p[1:]
+	}
```

Comment:
Should we not rely on [filepath.VolumeName](https://pkg.go.dev/path/filepath#VolumeName) instead?

Reply:
Maybe [filepath.Clean](https://pkg.go.dev/path/filepath#Clean) is all we need. I could not test it in Windows.

Reply:
+1 on the `filepath.Clean`.

Reply:
`filepath.Clean` changes `/` to `\` so that isn't ideal here - this is dealing with a URL path rather than a filesystem path.

As for `filepath.VolumeName`, I'm not sure that's appropriate either because it not only deals with drive letters, but also UNC paths like `\\host\share`.

Reply:
For what it's worth, I've tested changing this to `filepath.Clean` on a Windows VM and the test fails.

I also played around with `filepath.VolumeName` and this works:

```go
	if p != "" && p[0] == '/' && filepath.VolumeName(p[1:]) != "" {
		return p[1:]
	}
```

but it's less efficient than the existing `isDriveLetter` code (because it checks for UNC as well).

I'd love if this PR could just be merged as-is. I've been using a fork of this repo with this commit applied for the past 7 months without issue.

Reply:
IMO, the stdlib `filepath.VolumeName` would be better suited than having our own implementation. But the current state of this PR is working and resolves a legit problem users are having.

We can revisit this at a later stage, so will be merging this as-is.